### PR TITLE
Load runtime port configuration from http(s)-port file

### PR DESCRIPTION
**Pull request by andig**
This PR provides the runtime configuration for https://github.com/gokrazy/gokrazy/issues/74. 
The host part for creating the files from `gokr-packager` is still missing.

### CODE REVIEW

File: gokrazy.go [hunk 1]
```diff
@@ -104,6 +105,19 @@ func setupTLS() error {
 	return nil
 }
 
+// readFile reads configuration files from /perm /etc or / and returns trimmed content as string
+func readFile(fileName string) (string, error) {
```
Comment (stapelberg) [inline gokrazy.go:109]: Can you name this readConfigFile to be a little more descriptive please?

File: gokrazy.go [hunk 2]
```diff
@@ -223,7 +230,17 @@ func Supervise(commands []*exec.Cmd) error {
 		return err
 	}
 
-	if err := updateListenerPairs("80", "443", useTLS, tlsConfig); err != nil {
+	httpPort, err := readFile("http-port.txt")
+	if _, numeric := strconv.Atoi(httpPort); err != nil || numeric != nil {
+		httpPort = "80"
```
Comment (stapelberg) [inline gokrazy.go:235]: Here and below: please log an error message when the file is present but malformed, so that users know what the problem is.

Comment (andig) [reply gokrazy.go]: Split into function as the validation with logging is more than 2 lines

**Comment by andig** [comment]
I can confirm that with https://github.com/gokrazy/tools/commit/90d0f0f832e71126a70529ec51bd5dd6ef663e18 applied I can curl into the port specified with `http_port`.

If you want to merge and upgrade `gokrazy/tools` to use it, please upgrade `x/sys` there to latest as well or it won't build on OSX.

**Review by stapelberg — APPROVED** [review]

**Pull request merged**